packages/sdk/src/client/backend/metrom.ts

```diff
@@ -0,0 +1,149 @@
+import type { Address, PublicClient } from "viem";
+import { CoreClient } from "../core";
+import type { FetchCampaignsResponse, RawClaimableRewards } from "./types";
+import type { SupportedChain } from "@metrom-xyz/contracts";
+import { SUPPORTED_CHAIN_NAMES, SupportedAmm } from "../../commons";
+import type { Campaign, ClaimableRewards } from "../../entities";
+
+export type FetchClaimsParams = {
+    address: Address;
+    publicClient: PublicClient;
+};
+
+export type FetchCampaignsParams = {
+    pageNumber?: number;
+    pageSize?: number;
+    orderBy?: string;
+    orderDirection?: "asc" | "desc";
+};
+
+export type FetchCampaignsResult = {
+    campaigns: Campaign[];
+    amount: number;
+};
+
+export class MetromApiClient extends CoreClient {
+    private readonly targetChainName: string;
+
+    constructor(
+        public readonly baseUrl: string,
+        public readonly chain: SupportedChain,
+    ) {
+        super();
+
+        if (!SUPPORTED_CHAIN_NAMES[chain])
+            throw new Error(
+                `unsupported chain, supported chains are: ${Object.keys(SUPPORTED_CHAIN_NAMES)}`,
+            );
+
+        this.targetChainName = SUPPORTED_CHAIN_NAMES[chain];
+    }
+
+    async fetchCampaigns(
+        params?: FetchCampaignsParams,
+    ): Promise<FetchCampaignsResult> {
+        const queryParams: Record<string, string | number> = {
+            pageNumber: 0,
+            pageSize: 10,
+            orderBy: "creationTimestamp",
+            orderDirection: "desc",
+            ...Object.fromEntries(
+                Object.entries(params || {}).filter(
+                    ([_, value]) => !!value || !isNaN(Number(value)),
+                ),
+            ),
+        };
+
+        const url = new URL(`${this.targetChainName}/campaigns`, this.baseUrl);
+        Object.keys(queryParams).forEach((param) => {
+            url.searchParams.set(param, queryParams[param].toString());
+        });
+
+        const response = await fetch(url);
+        if (!response.ok)
+            throw new Error(
+                `response not ok while fetching campaigns: ${await response.text()}`,
+            );
+
+        const rawCampaignsResponse =
+            (await response.json()) as FetchCampaignsResponse;
+
+        return {
+            campaigns: rawCampaignsResponse.campaigns.map((rawCampaign) => ({
+                id: rawCampaign.id,
+                createdAt: rawCampaign.createdAt,
+                owner: rawCampaign.owner,
+                pendingOwner: rawCampaign.pendingOwner,
+                from: rawCampaign.from,
+                to: rawCampaign.to,
+                pair: {
```

Comment:
Can we call this pool? In the future we might support pools with more than 2 tokens. Also, is it really necessary to map everything? Can we maybe use the spread operator a bit more?

Reply:
Fixed in ca58144417aad83dacd68ed9df0bdfa97627b193.